File: server/database/scheam/category.js
import mongoose from 'mongoose'
const Schema  =mongoose.Schema
const Mixed =Schema.Types.Mixed
const ObjectId =Schema.Types.ObjectId

/**
 * 商品类型集合
 */
const CategorySchema = new Schema({
  name:String,
  children:[{
      type:ObjectId,
      ref:'ChildCateGory'
  }],
  image:String,
  meta:{
    createdAt: {
      type: Date,
      default: Date.now()
    },
    updatedAt: {
      type: Date,
      default: Date.now()
    }
  }
})

CategorySchema.pre('save',function(next){
    if(this.isNew){
        this.meta.createdAt = this.meta.updatedAt = Date.now()
    }else {
        this.meta.updatedAt = Date.now()
    }

    next()
})

mongoose.model('Category',CategorySchema)

File: minappServer/minappServer.Repositories/Entities/Category.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace minappServer.Repositories.Entities
{
    public class Category
    {
        public Category()
        {
            Children = new List<string>();
        }
        public string Name { get; set; }

        public List<string> Children { get; set; }
        public string image { get; set; }

    }
}
